File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        nums1.extend(nums2)
        nums1.sort()
        
        n = len(nums1)
        
        if n % 2 == 1:
            median = nums1[n // 2]
        else:
            mid1 = n // 2
            mid2 = mid1 - 1
            median = (nums1[mid1] + nums1[mid2]) / 2.0
        
        return median

solution = Solution()
nums1 = [1, 2]
nums2 = [3, 4]
res = solution.findMedianSortedArrays(nums1, nums2)
print(res)

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int[] merged=new int[nums1.length+nums2.length];
        int pos=0;
        for(int i=0;i<nums1.length;i++){
            merged[pos]=nums1[i];
            pos+=1;
        }

        for(int i=0;i<nums2.length;i++){
            merged[pos]=nums2[i];
            pos+=1;
        }

    int n = merged.length;
        for (int i = 1; i < n; ++i) {
            int key = merged[i];
            int j = i - 1;

            while (j >= 0 && merged[j] > key) {
                merged[j + 1] = merged[j];
                j = j - 1;
            }
            merged[j + 1] = key;
        }
        if (merged.length%2!=0){
            return merged[merged.length/2];
        }
        else{
            return ((float)(merged[merged.length/2]+merged[(merged.length/2)-1]))/2;
        }
    }
}